### PR TITLE
NV 2613 - Tenant Page Pagination

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Created the same pagination we have on the subscriber list page in order to reuse existing components.

### Why was this change needed?

in order to check long lists of tenants.

### Other information (Screenshots)

https://www.loom.com/share/f8e823495fe34c2d9fa542da79b60f0f

### CODE REVIEW

File: apps/api/src/app/shared/dtos/pagination-response.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { ApiProperty } from '@nestjs/swagger';
+import { IPaginatedResponseDto } from '@novu/shared';
 
-export class PaginatedResponseDto<T> {
+export class PaginatedResponseDto<T> implements IPaginatedResponseDto<T> {
```
Comment (djabarovgeorge) [inline apps/api/src/app/shared/dtos/pagination-response.ts:4]: added new Shared IPaginatedResponseDto as an interface for the API DTO

File: apps/web/src/api/tenants.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { api } from './api.client';
 
-export function getTenants() {
-  return api.get('/v1/tenants');
+export function getTenants({ page = 0, limit = 10 } = {}) {
+  return api.getFullResponse('/v1/tenants', { page, limit });
```
Comment (djabarovgeorge) [inline apps/web/src/api/tenants.ts:4]: changed to get the hasMore and other pagination data.

File: apps/web/src/hooks/useTenants.ts
Comment (djabarovgeorge) [inline apps/web/src/hooks/useTenants.ts:1]: refactored to get the full response of the pagination and return from the hook.

File: apps/web/src/hooks/useTenants.ts [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
-  const { data, isLoading, ...rest } = useQuery<ITenantEntity[], any, ITenantEntity[]>(
-    [QueryKeys.tenantsList],
-    getTenants,
-    { refetchOnMount: false, ...options }
+export function useTenants({
+  options,
+  page,
+  limit,
+}: {
+  options?: UseQueryOptions<IPaginatedResponseDto<ITenantEntity>, any>;
```
Comment (djabarovgeorge) [inline apps/web/src/hooks/useTenants.ts:14]: This is the place where i wanted to reuse the IPaginatedResponseDto interface.

File: apps/web/src/pages/tenants/components/list/TenantsList.tsx [hunk 4]
```diff
@@ -27,7 +34,19 @@ export const TenantsList = ({ onAddTenantClick }: { onAddTenantClick: React.Mous
...
+          data-test-id="tenants-list-table"
+          columns={columns}
+          data={tenants || []}
+          pagination={{
+            pageSize: pageSize,
+            current: page,
+            hasMore,
+            minimalPagination: true,
+            onPageChange: handleTableChange,
+          }}
```
Comment (djabarovgeorge) [inline apps/web/src/pages/tenants/components/list/TenantsList.tsx:48]: added the pagination prop data.

File: libs/shared/src/dto/shared/shared.ts
Comment (djabarovgeorge) [inline libs/shared/src/dto/shared/shared.ts:1]: new interface that i used in the useTenant hook and API DTO

**Review by ainouzgali — APPROVED** [review]
⭐️

**Pull request merged**